controller_manager/controller_manager/controller_manager_services.py

```diff
@@ -28,15 +28,17 @@
 import rclpy
 
 
-def service_caller(node, service_name, service_type, request, service_timeout=10.0):
+def service_caller(node, service_name, service_type, request, service_timeout=0.0):
     cli = node.create_client(service_type, service_name)
 
-    if not cli.service_is_ready():
-        node.get_logger().debug(
-            f"waiting {service_timeout} seconds for service {service_name} to become available..."
-        )
-        if not cli.wait_for_service(service_timeout):
-            raise RuntimeError(f"Could not contact service {service_name}")
+    while not cli.service_is_ready():
+        node.get_logger().debug(f"waiting for service {service_name} to become available...")
```

Comment:
This should probably be an info and not a debug. If the spawner hangs there users should know.

I tested this PR locally and managed to end up in a deadlock here, which is basically why I went for [this](https://github.com/ros-controls/ros2_control/pull/1501/files#diff-8b98f18ffd15d1897cb6658a1ca17bbfec87c4a9e61340dca22fb246da447d42R45-R55) approach in my PR. I realized that I can reproduce this quite regularly when restarting my robot bringup right after stopping it.

Reply:
I think we should also jump out after some number of attempts

Reply:
```suggestion
        node.get_logger().info(f"waiting for service {service_name} to become available...")
```
changing for an info

@fmauch after I merge this, could you please put up a new PR adding your solution to this including the max attempt logic?